### DISCUSSION TITLE
Legend Alignment in Charts

### DISCUSSION THREAD

**Discussion by rubensfernando** [vega/vega-lite#9233] [Q&A]
Hello everyone,

I hope you're all doing well.

I'm conducting some tests and would like to know if anyone has experience with a similar situation.

I'm working with a horizontal bar chart and have placed the legends at the top. Currently, with the settings orient="top" and direction="horizontal," the legends align with the Y-axis line. However, my goal is to align the legends with the left margin of the chart.

Here's an illustrative image showing the current alignment. If anyone has any tips or experience in adjusting the legend alignment in this way, I would greatly appreciate the help.

**Current version**

![image](https://github.com/vega/vega-lite/assets/1429909/6863b6cb-5ea4-4e33-b6cd-e3787f94c3b4)

**I would like it to look like this.**

Is it possible?

![image](https://github.com/vega/vega-lite/assets/1429909/247ec0e5-e456-4bd7-8b54-4543cbf66e45)

Thank you in advance for your assistance.

Edited:  I added the image of how I would like it to look.
Edited2: If I change the orientation to "none," can I automatically adjust the legend position for each chart without manually setting the values for legendX and legendY?

**Comment by ChiaLingWeng** [comment]
Maybe you can try to set orient to `None` and set `legendX: negative value`,
for example [Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEzjQATjRyZ289AEEABEyTrDsU5kP0kVqGwQ5Ta87fJtNLpFHVsIEawB3NkMATzhTCEpFEB1MQTg0ECMAZRpHBMMAIVNDAGEYSxjlWxRUUGIkQQY1NABtUDy3Mg9QpP0AGQ6YgE0I9TQAJgAGQYBmJQBVWToIIYB2YfmANgBfBUbm1vau3v6h0YAWKZnMOdQxS4AOQ-XNzHd1NvQ8gHU90wPxk9m0ZcOxrcNiAmg8Wk8km8PgNUCNBscQNNfqgAIxogCcwzuIK2EPQADkAPLQr4TRGnc6DZY3bGgx7PEBEkmwo4-M5DQ5iIH3elJfHvJR9T4s77k5GDeaY2m4hn85lwhFI9mow7zQbSsHbdApACi8tGZKV5xRhxRY3VwLp4IZuv18LZlPRYiWGt52uJgv2IsNFLQKOGYzWlplSRSHpAQphCodfuG6OWKNWAF11iBkOoANZJEwDJRwWT2ZQ0WRkNCgAAeZZAADMaHBBMokkaYphQjhEugAI4MJCyOK2GikGJIcs0c6gOIJJIgVap56gWv1xsvENKVvtpKyBzFqrD0fj2J0KfoJUto9wfQFmAeJIQTBFNfny9kY9Y1P2QQ38o1usNpKRls2w7EAtwQHdBBnJRQkJatqwgOBMCrRc-3QADZyUexZFrUtv0nDsFzkTA0gALw7E0lF7WAvxAO8HxADw6z7JJMDYHBIJABIyHzZdQAY-NEPQLdZESR94g7WQGEEQQlE47iAA00H4MZhhk8huJ6NA31WIA)
